### PR TITLE
Manually-triggered releases, and reduce number of checks

### PR DESCRIPTION
**Pull request by Wulf**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -1,13 +1,16 @@
-name: Continuous Integration
-on: 
+name: CI
+on:
   push:
+    branches:
+      - main
```
Comment (hasezoey) [inline .github/workflows/CI.yml:5]: why only limit CI to branch `main`? (release stage is already limited)

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -92,15 +94,15 @@ jobs:
...
       - run: rustup component add rustfmt
       - run: cargo fmt --all -- --check
-    
+
   release:
     runs-on: ubuntu-latest
     name: release
     needs: [build, clippy, test, test-script, fmt]
-    if: github.ref == 'refs/heads/release'
+    if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main'
```
Comment (hasezoey) [inline .github/workflows/CI.yml:102]: why did `release` change to `main`, did the release strategy change?

**Review by hasezoey — COMMENTED** [review]
i am not quite sure what this PR is trying to do, but i have some questions

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Wulf) [reply .github/workflows/CI.yml:5]: It's not just limited to `main` :)

It also runs for PRs!

File: .github/workflows/CI.yml (hunk 2 above)
Comment (Wulf) [reply .github/workflows/CI.yml:102]: yeah, exactly. I was looking at your concerns in #71 and wanted to make things a bit more simpler!

File: .github/workflows/CI.yml (hunk 1 above)
Comment (hasezoey) [reply .github/workflows/CI.yml:5]: yes the CI will run on *PR*s, but from what i can tell, it wont run on other branches in this repository

**Pull request merged**